libraries/microsoft-agents-hosting-core/microsoft_agents/hosting/core/app/agent_application.py

```diff
@@ -738,7 +779,7 @@ async def _run_after_turn_middleware(self, context: TurnContext, state: StateT):
         return True
 
     async def _on_activity(self, context: TurnContext, state: StateT):
-        for route in self._routes:
+        for route in self._route_list:
```

Comment:
Suggest delegating this to the route list, so that concurrency is easier managed. I do realize that's already using a custom iterator that's cloning the list, but reading the code it's really hard to see that clearly.